src/Parsers/Access/ParserCreateRoleQuery.cpp

```diff
@@ -91,6 +91,10 @@ bool ParserCreateRoleQuery::parseImpl(Pos & pos, ASTPtr & node, Expected & expec
     String new_name;
     std::shared_ptr<ASTSettingsProfileElements> settings;
     String cluster;
+    String storage_name;
+
+    if (ParserKeyword{"AT"}.ignore(pos, expected))
+        parseStorageName(pos, expected, storage_name);
```

Comment:
Maybe it's better not to fix the position of the `AT` keyword and move that condition into the `while` cycle below?

Reply:
The statement can look a bit weird:

```
CREATE USER blabla@remote AT local
```